test/functional/keyed_items.ts

```diff
@@ -413,3 +426,48 @@ describe('getKey', () => {
     expect(getKey(div.firstChild!)).to.equal('3')
   });
 });
+
+describe('setKeyAttributeName', () => {
+  let container: HTMLElement;
+  let keyedEl: HTMLElement;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    keyedEl = document.createElement('div');
+    keyedEl.setAttribute('key', 'foo');
+    keyedEl.setAttribute('secondaryKey', 'bar');
+    container.appendChild(keyedEl);
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+    setKeyAttributeName('key'); // Default.
+  });
+
+  it('should use the default `key` attribute', () => {
+    patch(container, () => {
+      elementVoid('div', 'baz');
+    });
+    expect(getKey(keyedEl)).to.equal('foo')
```

Comment:
Would also add `expect(keyedEl.parentNode).to.be.null;` to make it clear that the element was imported, then replaced with the new div with a different key.